manual/trees/index.md

```diff
@@ -7,178 +7,145 @@ toc: true
 
 > **RNTuple**
 >
-> [RNTuple](https://root.cern/doc/master/md_tree_ntuple_v7_doc_README.html){:target="_blank"} (for N-tuple and nested tuple) is the experimental evolution of {% include ref class="TTree" %} columnar data storage. `RNTuple` introduces new interfaces that are more robust.
+> [RNTuple](https://root.cern/doc/master/md_tree_ntuple_v7_doc_README.html){:target="_blank"} is the experimental evolution of {% include ref class="TTree" %} columnar data storage. {% include ref class="RNTuple" namespace="ROOT::Experimental" %} introduces robust interfaces, a high-performance storage layout, and an asynchronous, thread-safe scheduling.
 
-## Tree classes
-
-ROOT provides numerous classes for trees and branches, of which the following are among the most used:
-
-- [TTree](https://root.cern/doc/master/classTTree.html){:target="_blank"}: Represents a columnar data set. Any C++ type can be stored in its columns.
+> **RDataFrame**
+>
+> To access TTree data, please use {% include ref class="RDataFrame" namespace="ROOT" %}.
+> `TTree` provides interfaces for low-level, expert usage.
 
-- [TNtuple](https://root.cern/doc/master/classTNtuple.html){:target="_blank"}: A simple `TTree` restricted to a list of float variables only.
+### The tree and its data
 
-- [TBranch](https://root.cern/doc/master/classTBranch.html){:target="_blank"}: Organizes columns, i.e. branches, of a `TTree`.
+A `TTree` behaves like an array of a data structure that resides on storage - except for one entry (or row, in database language).
+That entry is accessible in memory: you can load any tree entry, ideally sequentially.
+You can provide your own storage for the values of the columns of the current entry, in the form of variables.
+In this case you have to tell the `TTree` about the addresses of these variables; either by calling [`TTree::SetBranchAddress()`](https://root.cern/doc/master/classTTree.html#a39b867210e4a77ef44917fd5e7898a1d), or by passing the variable when creating the branch for writing.
+When "filling" (writing) the `TTree`, it will read the values out of these variables;
+when reading back a `TTree` entry, it will write the values it read from storage into your variables.
 
-- [TChain](https://root.cern/doc/master/classTChain.html){:target="_blank"}: A list of ROOT files containing `TTree` objects.
+### Branches and leaves
 
+A tree consists of a list of independent columns, called branches. A branch can contain values of any fundamental type, C++ objects known to ROOT's type system, or collections of those.
+Branches are represented by {% include ref class="TBranch" %} and its derived classes.
 
-## Working with trees
+While `TBranch` represent structure, objects inheriting from {% include ref class="TLeaf" %} contain the actual data.
+Originally, any columnar data was contained in a `TLeaf`; these days, some of the `TBranch`-derived classes contain data themselves, such as {% include ref class="TBranchElement" %}.
```

Comment:
> `contain data`

Might be an over simplifications.  Literally speaking the only entity that contains the data  are:
* the TBasket (in memory, uncompressed)
* the region of TFile corresponding to this TBasket (on disk, compressed)
* the user variable (in memory.

And it is always the TBranch that holds the information about the baskets.

The distinction between branches and leaves is "whom" hold the address of the memory corresponding to the user variable that hold the data or "whom" can give access to the user variable address and type information (`GetAddress`, `GetValue`).

so maybe `give access to the data` is more accurate.

Reply:
Done!